www/search.spt

```diff
@@ -15,12 +15,23 @@ if query:
 
     if scope in (None, 'usernames'):
         results['usernames'] = website.db.all("""
-            SELECT username, avatar_url, similarity(username, %(q)s) AS rank, id
+            SELECT username
+                 , public_name
+                 , avatar_url
+                 , CASE 
+                        WHEN public_name IS NULL 
+                             THEN similarity(username, %(q)s)
+                        WHEN similarity(username, %(q)s) >= similarity(public_name, %(q)s) 
+                             THEN similarity(username, %(q)s)
+                        ELSE similarity(public_name, %(q)s)
+                    END AS rank
+                 , id
               FROM participants
-             WHERE status = 'active'
-               AND NOT username like '~%%'
-               AND lower(username) %% lower(%(q)s)
+             WHERE lower(username) %% lower(%(q)s)
+                OR lower(public_name) %% lower(%(q)s)
+               AND status = 'active'
                AND hide_from_search = 0
+               AND NOT username like '~%%'
```

Comment:
This is incorrect due to the order of precedence of the boolean operators. `… OR … AND …` is equivalent to `… OR ( … AND … )`, but here you wanted `( … OR … ) AND …`.

Also, you forgot to add `public_name IS NOT NULL`, so the query isn't using the new index.

```suggestion
             WHERE status = 'active'
               AND hide_from_search = 0
               AND ( NOT username like '~%%' AND lower(username) %% lower(%(q)s)
                     OR
                     public_name IS NOT NULL AND lower(public_name) %% lower(%(q)s)
                   )
```